devops/scripts/benchmarks/html/chart-annotations.js

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2025 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+/**
+ * Find version changes in data points to create annotations
+ * @param {Array} points - Data points to analyze
+ * @param {string} versionKey - Key to track version changes
+ * @returns {Array} - List of change points
+ */
+function findVersionChanges(points, versionKey) {
+    if (!points || points.length < 2) return [];
+
+    const changes = [];
+    // Sort points by date
+    const sortedPoints = [...points].sort((a, b) => a.x - b.x);
+    let lastVersion = sortedPoints[0][versionKey];
+
+    for (let i = 1; i < sortedPoints.length; i++) {
+        const currentPoint = sortedPoints[i];
+
+        const currentVersion = currentPoint[versionKey];
+        if (currentVersion && currentVersion !== lastVersion) {
+            changes.push({
+                date: currentPoint.x,
+                newVersion: currentVersion,
+            });
+            lastVersion = currentVersion;
+        }
+    }
+    
+    return changes;
+}
+
+/**
+ * Add version change annotations to chart options
+ * @param {Object} data - Chart data
+ * @param {Object} options - Chart.js options object
+ */
+function addVersionChangeAnnotations(data, options) {
+    const changeTrackers = [
+        {
+            // Benchmark repos updates
+            versionKey: 'gitBenchHash',
+            sources: [
+                {
+                    name: "Compute Benchmarks",
+                    url: "https://github.com/intel/compute-benchmarks.git",
```

Comment:
aren't these stored in group metadata? we should ideally create this list automatically.

Reply:
done